### PR TITLE
HTTPS Target Type & Stealth Scan

### PR DESCRIPTION
**Pull request by VictorSuraj**
Describe your changes or your new module and let us know which OS and Python version you tested on.
_________________
**OS**: `x`

**OS Version**: `x`

**Python Version**: `x`

### CODE REVIEW

**Comment by VictorSuraj** [comment]
here target on port 443 as well as 80 is same and = HTTP

**Comment by pradeepjairamani** [comment]
What if the target is http then tarhet will become https://+ http:// + url that will result in an error

**Comment by Ali-Razmjoo** [comment]
Hello,

Thanks for your contribution, I think we need to rewrite something in there, and fix the modules, I had a problem with the target type too.

```
$ python nettacker.py -i https://offsecresearch.com -m port_scan
exit successfully.

$ python nettacker.py -i https://offsecresearch.com/ -m port_scan
[X] unknown type of target [https://offsecresearch.com/]
```

I think we need to create a UML for this issue and fix it for all kind of targets and false positive and false negative.

Regards.

**Comment by Ali-Razmjoo** [comment]
Hey @VictorSuraj,

Thanks a lot for contributing, let me check the code and get back to you soon.

Regards.

**Comment by VictorSuraj** [comment]
I can improve this code in much better form but some time required.

File: lib/scan/port/engine.py [hunk 1]
```diff
@@ -123,6 +124,22 @@ def extra_requirements_dict():
...
+			try:
+				if s[0][1].seq==0:
+					pass
+			except:
+				s = sr1(IP(dst=ip)/TCP(dport=0,flags='S'),timeout=2,verbose=0)
+				if s==None:
+					return None
+				else:
+					return 'port '+str(port)+' is FILTERED '
+	except:
```
[the remark this answers is not in the document]
Comment (Ali-Razmjoo) [reply lib/scan/port/engine.py]: it's not necessary to fix this, I will ignore it and add a new task to fix all `try, except, pass` in the project.

File: lib/scan/port/engine.py [hunk 2]
```diff
@@ -123,6 +124,22 @@ def extra_requirements_dict():
     conf.verb = 0
     conf.nofilter = 1
 
+def filter(ip,port):
```
[the remark this answers is not in the document]
Comment (Ali-Razmjoo) [reply lib/scan/port/engine.py]: you may rename this function. thanks.

**Comment by Ali-Razmjoo** [comment]
@VictorSuraj sure take your time.

**Comment by shaddygarg** [comment]
@VictorSuraj, your spacing inside the  `engine.py` is incosistent with the whole project. You should set your tabspace to 4 instead of 8.

File: lib/scan/port/engine.py [hunk 3]
```diff
@@ -123,6 +121,22 @@ def extra_requirements_dict():
...
+            try:
+                if s[0][1].seq==0:
+                    pass
+            except:
+                s = sr1(IP(dst=ip)/TCP(dport=0,flags='S'),timeout=2,verbose=0)
+                if s==None:
+                    return None
+                else:
+                    return 'port '+str(port)+' is FILTERED '
+    except:
```
[the remark this answers is not in the document]
Comment (VictorSuraj) [reply lib/scan/port/engine.py]: I cant understand what is mean of this issue. In my PC the output comes in form of this.
show below in files
port

[the remark this answers is not in the document]
Comment (VictorSuraj) [reply lib/scan/port/engine.py]: final_port
this is filterd port scanning view

**Comment by Ali-Razmjoo** [comment]
Hey,

As you sent two PRs in once, I should add a note in here to fix url detecting later, it still has some errors.

Regards.

**Comment by Ali-Razmjoo** [comment]
Hey,

I merged the PR, thanks for your contribution, Also I should mention that you can take a look at wiki [developers page](https://github.com/zdresearch/OWASP-Nettacker/wiki/Developers) ( I updated it) to find out how you can contribute more if you are interested.

Best Regards.

**Comment by VictorSuraj** [comment]
Thanks sir for your guidance I will try my best.

**Pull request merged**